enterprise/cypher/cypher/src/main/scala/org/neo4j/cypher/internal/BuildSlottedExecutionPlan.scala

```diff
@@ -86,8 +86,10 @@ object BuildSlottedExecutionPlan extends Phase[EnterpriseRuntimeContext, Logical
       val executionPlanBuilder = new PipeExecutionPlanBuilder(context.clock, context.monitors,
                                                               expressionConverters = converters,
                                                               pipeBuilderFactory = pipeBuilderFactory)
+      val readOnlies = new ReadOnlies
```

Comment:
We don't need an Attribute to keep track of read only plans. We just need to know if the whole query is read-only.